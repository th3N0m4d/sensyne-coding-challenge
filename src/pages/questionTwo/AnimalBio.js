import React from 'react'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import CardMedia from '@material-ui/core/CardMedia'
import Typography from '@material-ui/core/Typography'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'

import styles from './styles'

const propTypes = {
  title: PropTypes.string,
  imgSrc: PropTypes.any,
  body: PropTypes.any,
  classes: PropTypes.object
}

const defaultProps = {
  title: '',
  imgSrc: null,
  body: '',
  classes: {}
}

const AnimalBio = ({ title, imgSrc, body, classes }) => (
  <div className={classes.container}>
    <Card className={classes.card}>
      {
        imgSrc &&
          <CardMedia
            className={classes.media}
            image={imgSrc}
            title={title}
          />
      }
      <CardContent className={classes.content}>
        <Typography gutterBottom variant='h5' component='h2'>
          {title}
        </Typography>
        <div
          className={classes.body}
          dangerouslySetInnerHTML={{ __html: body }}
        />
      </CardContent>
    </Card>
  </div>
)

AnimalBio.propTypes = propTypes
AnimalBio.defaultProps = defaultProps

export default withStyles(styles)(AnimalBio)
