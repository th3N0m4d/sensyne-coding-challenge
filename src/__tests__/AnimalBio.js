import React from 'react'
import renderer from 'react-test-renderer'

import AnimalBio from '../pages/questionTwo/AnimalBio'

const expectToMatchSnapshot = component => {
  expect(renderer.create(component).toJSON()).toMatchSnapshot()
}

describe('AnimalBio', () => {
  it('should render component', () => {
    expectToMatchSnapshot(<AnimalBio />)
  })

  it('should render component with title', () => {
    expectToMatchSnapshot(<AnimalBio title='Hello there!' />)
  })

  it('should render component with body', () => {
    const body = `
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.      
      `

    expectToMatchSnapshot(<AnimalBio body={body} />)
  })

  it('should render image', () => {
    const imgSrc = 'data:image/png;base64'

    expectToMatchSnapshot(<AnimalBio imgSrc={imgSrc} />)
  })
})
