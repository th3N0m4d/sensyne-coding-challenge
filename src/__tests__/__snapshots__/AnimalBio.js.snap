// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimalBio should render component 1`] = `
<div
  className="AnimalBio-container-1"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiCard-root AnimalBio-card-2 MuiPaper-rounded"
  >
    <div
      className="MuiCardContent-root AnimalBio-content-4"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        
      </h2>
      <div
        className="AnimalBio-body-5"
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`AnimalBio should render component with body 1`] = `
<div
  className="AnimalBio-container-1"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiCard-root AnimalBio-card-2 MuiPaper-rounded"
  >
    <div
      className="MuiCardContent-root AnimalBio-content-4"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        
      </h2>
      <div
        className="AnimalBio-body-5"
        dangerouslySetInnerHTML={
          Object {
            "__html": "
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.      
      ",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`AnimalBio should render component with title 1`] = `
<div
  className="AnimalBio-container-1"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiCard-root AnimalBio-card-2 MuiPaper-rounded"
  >
    <div
      className="MuiCardContent-root AnimalBio-content-4"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Hello there!
      </h2>
      <div
        className="AnimalBio-body-5"
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`AnimalBio should render image 1`] = `
<div
  className="AnimalBio-container-1"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiCard-root AnimalBio-card-2 MuiPaper-rounded"
  >
    <div
      className="MuiCardMedia-root AnimalBio-media-3"
      style={
        Object {
          "backgroundImage": "url(\\"data:image/png;base64\\")",
        }
      }
      title=""
    />
    <div
      className="MuiCardContent-root AnimalBio-content-4"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        
      </h2>
      <div
        className="AnimalBio-body-5"
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
    </div>
  </div>
</div>
`;
